### PR TITLE
Django exercise

### PR DESCRIPTION
**Pull request by Taiters**
Code written while going through the Django onboarding exercise / Udemy course. Sorry for the giant PR, I would normally avoid this.

This includes the requirements from the onboarding docs (recipes & ingredients) as well as tags, filtering and authentication functionality from the Django exercise

### CODE REVIEW

File: app/core/tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        self.assertTrue(user.is_superuser)
+        self.assertTrue(user.is_staff)
+
+    def test_create_recipe(self):
+        """Test creating a recipe is successful."""
+        user = get_user_model().objects.create_user(
+            "test@example.com",
+            "testpass123",
+        )
```
Comment (Thepnathi) [inline app/core/tests/test_models.py:65]:
```suggestion
user = create_user()
```

**Review by Thepnathi — APPROVED** [review]
LGTM 👍 
Some of the things I really like:
* Added small comments for each test
* Covered a lot of edge cases

Just minor thing I seen and would suggest you split up the PR by each model and maybe use test [fixtures](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html ) to avoid duplication like creating user object.

File: app/core/tests/test_models.py (hunk 1 above)
Comment (Taiters) [reply app/core/tests/test_models.py:65]: Well spotted, thanks :)
